error: unreachable expression
  --> $DIR/expr_assign.rs:19:5: in fn foo
   |
LL |     x = return; //~ ERROR unreachable
   |     ^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/expr_assign.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_assign.rs:29:14: in fn bar
   |
LL |         *p = return; //~ ERROR unreachable
   |              ^^^^^^

error: unreachable expression
  --> $DIR/expr_assign.rs:35:15: in fn baz
   |
LL |     *{return; &mut i} = 22; //~ ERROR unreachable
   |               ^^^^^^

error: aborting due to 3 previous errors

