error: unreachable expression
  --> $DIR/expr_method.rs:25:21: in fn a
   |
LL |     Foo.foo(return, 22); //~ ERROR unreachable
   |                     ^^
   |
note: lint level defined here
  --> $DIR/expr_method.rs:14:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_method.rs:30:5: in fn b
   |
LL |     Foo.bar(return); //~ ERROR unreachable
   |     ^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

