error[E0594]: cannot assign to `x.0`, as `x` is not declared as mutable
  --> $DIR/reassignment_immutable_fields_twice.rs:7:5
   |
LL |     let x: (u32, u32);
   |         - help: consider changing this to be mutable: `mut x`
LL |     x = (22, 44);
LL |     x.0 = 1; //~ ERROR
   |     ^^^^^^^ cannot assign

error[E0594]: cannot assign to `x.0`, as `x` is not declared as mutable
  --> $DIR/reassignment_immutable_fields_twice.rs:12:5
   |
LL |     let x: (u32, u32);
   |         - help: consider changing this to be mutable: `mut x`
LL |     x.0 = 1; //~ ERROR
   |     ^^^^^^^ cannot assign

error[E0594]: cannot assign to `x.0`, as `x` is not declared as mutable
  --> $DIR/reassignment_immutable_fields_twice.rs:13:5
   |
LL |     let x: (u32, u32);
   |         - help: consider changing this to be mutable: `mut x`
LL |     x.0 = 1; //~ ERROR
LL |     x.0 = 22; //~ ERROR
   |     ^^^^^^^^ cannot assign

error[E0594]: cannot assign to `x.1`, as `x` is not declared as mutable
  --> $DIR/reassignment_immutable_fields_twice.rs:14:5
   |
LL |     let x: (u32, u32);
   |         - help: consider changing this to be mutable: `mut x`
...
LL |     x.1 = 44; //~ ERROR
   |     ^^^^^^^^ cannot assign

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0594`.
