error[E0597]: borrowed value does not live long enough
  --> $DIR/const-ptr-nonnull.rs:4:37
   |
LL |     let x: &'static NonNull<u32> = &(NonNull::dangling()); //~ ERROR borrowed value does not live long enough
   |                                     ^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
...
LL | }
   | - temporary value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: borrowed value does not live long enough
  --> $DIR/const-ptr-nonnull.rs:8:37
   |
LL |     let x: &'static NonNull<u32> = &(non_null.cast()); //~ ERROR borrowed value does not live long enough
   |                                     ^^^^^^^^^^^^^^^^^ temporary value does not live long enough
LL | }
   | - temporary value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0597`.
