error[E0658]: renaming extern crates with `_` is unstable (see issue #48216)
  --> $DIR/feature-gate-underscore-imports.rs:1:1
   |
LL | extern crate std as _; //~ ERROR renaming extern crates with `_` is unstable
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(underscore_imports)] to the crate attributes to enable

error[E0658]: renaming imports with `_` is unstable (see issue #48216)
  --> $DIR/feature-gate-underscore-imports.rs:2:5
   |
LL | use std::vec as _; //~ ERROR renaming imports with `_` is unstable
   |     ^^^^^^^^^^^^^
   |
   = help: add #![feature(underscore_imports)] to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
