error: `#[proc_macro]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:9:1
   |
LL | #[proc_macro = "test"] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^^^^^^^^

error: `#[proc_macro]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:12:1
   |
LL | #[proc_macro()] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^

error: `#[proc_macro]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:15:1
   |
LL | #[proc_macro(x)] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^^

error: `#[proc_macro_attribute]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:18:1
   |
LL | #[proc_macro_attribute = "test"] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[proc_macro_attribute]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:21:1
   |
LL | #[proc_macro_attribute()] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[proc_macro_attribute]` attribute does not take any arguments
  --> $DIR/invalid-attributes.rs:24:1
   |
LL | #[proc_macro_attribute(x)] //~ ERROR: does not take any arguments
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

