error[E0425]: cannot find function `wants_clone` in this scope
  --> $DIR/universal_wrong_bounds.rs:16:5
   |
LL |     wants_clone(f); //~ ERROR cannot find
   |     ^^^^^^^^^^^ did you mean `wants_cone`?

error[E0405]: cannot find trait `Debug` in this scope
  --> $DIR/universal_wrong_bounds.rs:19:24
   |
LL | fn wants_debug(g: impl Debug) { } //~ ERROR cannot find
   |                        ^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL | use std::fmt::Debug;
   |

error[E0405]: cannot find trait `Debug` in this scope
  --> $DIR/universal_wrong_bounds.rs:20:26
   |
LL | fn wants_display(g: impl Debug) { } //~ ERROR cannot find
   |                          ^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
LL | use std::fmt::Debug;
   |

error: cannot continue compilation due to previous error

