error[E0495]: cannot infer an appropriate lifetime for borrow expression due to conflicting requirements
  --> $DIR/issue-40510-1.rs:20:9
   |
LL |         &mut x //~ ERROR cannot infer
   |         ^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime  as defined on the body at 19:5...
  --> $DIR/issue-40510-1.rs:19:5
   |
LL |     || {
   |     ^^
note: ...so that closure can access `x`
  --> $DIR/issue-40510-1.rs:20:9
   |
LL |         &mut x //~ ERROR cannot infer
   |         ^^^^^^
note: but, the lifetime must be valid for the expression at 19:5...
  --> $DIR/issue-40510-1.rs:19:5
   |
LL | /     || {
LL | |         &mut x //~ ERROR cannot infer
LL | |     };
   | |_____^
note: ...so type `[closure@$DIR/issue-40510-1.rs:19:5: 21:6 x:&mut std::boxed::Box<()>]` of expression is valid during the expression
  --> $DIR/issue-40510-1.rs:19:5
   |
LL | /     || {
LL | |         &mut x //~ ERROR cannot infer
LL | |     };
   | |_____^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0495`.
