error[E0598]: lifetime of `x` is too short to guarantee its contents can be safely reborrowed
  --> $DIR/issue-40510-3.rs:20:9
   |
LL |         || { //~ ERROR too short
   |         ^^
   |
note: `x` would have to be valid for the expression at 19:5...
  --> $DIR/issue-40510-3.rs:19:5
   |
LL | /     || {
LL | |         || { //~ ERROR too short
LL | |             x.push(())
LL | |         }
LL | |     };
   | |_____^
note: ...but `x` is only valid for the lifetime  as defined on the body at 19:5
  --> $DIR/issue-40510-3.rs:19:5
   |
LL |     || {
   |     ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0598`.
