error: unreachable expression
  --> $DIR/unwarned-match-on-never.rs:14:5
   |
LL |     match never() {} //~ ERROR unreachable expression
   |     ^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/unwarned-match-on-never.rs:3:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable arm
  --> $DIR/unwarned-match-on-never.rs:19:15
   |
LL |         () => () //~ ERROR unreachable arm
   |               ^^

error: unreachable expression
  --> $DIR/unwarned-match-on-never.rs:25:5
   |
LL | /     match () { //~ ERROR unreachable expression
LL | |         () => (),
LL | |     }
   | |_____^

error: aborting due to 3 previous errors

