error[E0005]: refutable pattern in local binding: `_` not covered
  --> $DIR/const-pattern-irrefutable.rs:22:9: in fn main
   |
LL |     let a = 4; //~ ERROR refutable pattern in local binding: `_` not covered
   |         ^ interpreted as a constant pattern, not new variable

error[E0005]: refutable pattern in local binding: `_` not covered
  --> $DIR/const-pattern-irrefutable.rs:23:9: in fn main
   |
LL |     let c = 4; //~ ERROR refutable pattern in local binding: `_` not covered
   |         ^ interpreted as a constant pattern, not new variable

error[E0005]: refutable pattern in local binding: `_` not covered
  --> $DIR/const-pattern-irrefutable.rs:24:9: in fn main
   |
LL |     let d = 4; //~ ERROR refutable pattern in local binding: `_` not covered
   |         ^ interpreted as a constant pattern, not new variable

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0005`.
