error[E0308]: mismatched types
  --> $DIR/existential_type.rs:18:5
   |
LL | fn foo() -> Foo { //~ ERROR trait bound `impl ?Sized: std::marker::Sized` is not satisfied
   |             --- expected `impl ?Sized` because of return type
LL |     "" //~ ERROR mismatched types
   |     ^^ expected anonymized type, found reference
   |
   = note: expected type `impl ?Sized`
              found type `&'static str`

error[E0277]: the trait bound `impl ?Sized: std::marker::Sized` is not satisfied
  --> $DIR/existential_type.rs:17:13
   |
LL | fn foo() -> Foo { //~ ERROR trait bound `impl ?Sized: std::marker::Sized` is not satisfied
   |             ^^^ `impl ?Sized` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `impl ?Sized`
   = note: the return type of a function must have a statically known size

error: aborting due to 2 previous errors

Some errors occurred: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
