error[E0705]: types cannot be named `dyn`
  --> $DIR/issue-50405-types-named-dyn.rs:15:5
   |
LL | /     trait dyn { //~ ERROR types cannot be named `dyn`
LL | |     }
   | |_____^

error[E0705]: types cannot be named `dyn`
  --> $DIR/issue-50405-types-named-dyn.rs:20:5
   |
LL |     struct dyn { } //~ ERROR types cannot be named `dyn`
   |     ^^^^^^^^^^^^^^

error[E0705]: types cannot be named `dyn`
  --> $DIR/issue-50405-types-named-dyn.rs:24:5
   |
LL |     enum dyn { } //~ ERROR types cannot be named `dyn`
   |     ^^^^^^^^^^^^

error[E0705]: types cannot be named `dyn`
  --> $DIR/issue-50405-types-named-dyn.rs:28:5
   |
LL |     union dyn { _f: () } //~ ERROR types cannot be named `dyn`
   |     ^^^^^^^^^^^^^^^^^^^^

error[E0705]: types cannot be named `dyn`
  --> $DIR/issue-50405-types-named-dyn.rs:32:5
   |
LL |     type dyn = u32; //~ ERROR types cannot be named `dyn`
   |     ^^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0705`.
