error[E0391]: cycle detected when const-evaluating `X::A::{{constant}}`
  --> $DIR/issue-23302-1.rs:14:9
   |
LL |     A = X::A as isize, //~ ERROR E0391
   |         ^^^^^^^^^^^^^
   |
note: ...which requires processing `X::A::{{constant}}`...
  --> $DIR/issue-23302-1.rs:14:9
   |
LL |     A = X::A as isize, //~ ERROR E0391
   |         ^^^^^^^^^^^^^
   = note: ...which again requires const-evaluating `X::A::{{constant}}`, completing the cycle

error: aborting due to previous error

For more information about this error, try `rustc --explain E0391`.
