error[E0695]: unlabeled `break` inside of a labeled block
  --> $DIR/label_break_value_unlabeled_break.rs:16:9: in fn unlabeled_break_simple
   |
LL |         break; //~ ERROR unlabeled `break` inside of a labeled block
   |         ^^^^^ `break` statements that would diverge to or through a labeled block need to bear a label

error[E0695]: unlabeled `break` inside of a labeled block
  --> $DIR/label_break_value_unlabeled_break.rs:24:13: in fn unlabeled_break_crossing
   |
LL |             break; //~ ERROR unlabeled `break` inside of a labeled block
   |             ^^^^^ `break` statements that would diverge to or through a labeled block need to bear a label

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0695`.
