warning: label name `'fl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:20:5: in fn foo
   |
LL |     'fl: for _ in 0..10 { break; }
   |     --- first declared here
LL |     'fl: loop { break; }           //~ WARN label name `'fl` shadows a label name that is already in scope
   |     ^^^ lifetime 'fl already in scope

warning: label name `'lf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:23:5: in fn foo
   |
LL |     'lf: loop { break; }
   |     --- first declared here
LL |     'lf: for _ in 0..10 { break; } //~ WARN label name `'lf` shadows a label name that is already in scope
   |     ^^^ lifetime 'lf already in scope

warning: label name `'wl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:25:5: in fn foo
   |
LL |     'wl: while 2 > 1 { break; }
   |     --- first declared here
LL |     'wl: loop { break; }           //~ WARN label name `'wl` shadows a label name that is already in scope
   |     ^^^ lifetime 'wl already in scope

warning: label name `'lw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:27:5: in fn foo
   |
LL |     'lw: loop { break; }
   |     --- first declared here
LL |     'lw: while 2 > 1 { break; }    //~ WARN label name `'lw` shadows a label name that is already in scope
   |     ^^^ lifetime 'lw already in scope

warning: label name `'fw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:29:5: in fn foo
   |
LL |     'fw: for _ in 0..10 { break; }
   |     --- first declared here
LL |     'fw: while 2 > 1 { break; }    //~ WARN label name `'fw` shadows a label name that is already in scope
   |     ^^^ lifetime 'fw already in scope

warning: label name `'wf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:31:5: in fn foo
   |
LL |     'wf: while 2 > 1 { break; }
   |     --- first declared here
LL |     'wf: for _ in 0..10 { break; } //~ WARN label name `'wf` shadows a label name that is already in scope
   |     ^^^ lifetime 'wf already in scope

warning: label name `'tl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:33:5: in fn foo
   |
LL |     'tl: while let Some(_) = None::<i32> { break; }
   |     --- first declared here
LL |     'tl: loop { break; }           //~ WARN label name `'tl` shadows a label name that is already in scope
   |     ^^^ lifetime 'tl already in scope

warning: label name `'lt` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:35:5: in fn foo
   |
LL |     'lt: loop { break; }
   |     --- first declared here
LL |     'lt: while let Some(_) = None::<i32> { break; }
   |     ^^^ lifetime 'lt already in scope

error: compilation successful
  --> $DIR/loops-reject-duplicate-labels.rs:49:1
   |
LL | / pub fn main() { //~ ERROR compilation successful
LL | |     let s = S;
LL | |     s.m1();
LL | |     s.m2();
LL | |     foo();
LL | | }
   | |_^

error: aborting due to previous error

