error: incorrect close delimiter: `}`
  --> $DIR/parser-recovery-4.rs:20:1
   |
LL | fn main() {
   |           - close delimiter possibly meant for this
LL |     let x = 1;
LL |     foo(x
   |        - un-closed delimiter
...
LL | }
   | ^ incorrect close delimiter

error: expected one of `!`, `)`, `,`, `.`, `::`, `?`, `{`, or an operator, found `bar`
  --> $DIR/parser-recovery-4.rs:17:5
   |
LL |     foo(x
   |        - -
   |        | |
   |        | expected one of 8 possible tokens here
   |        | help: ...the missing `)` may belong here
   |        if you meant to close this...
LL |     bar()
   |     ^^^ unexpected token

error[E0061]: this function takes 1 parameter but 2 parameters were supplied
  --> $DIR/parser-recovery-4.rs:16:5
   |
LL |   fn foo(_: usize) {}
   |   ---------------- defined here
...
LL | /     foo(x
LL | |     bar()
LL | |     //~^ ERROR expected one of
LL | |     //~^^^ ERROR this function takes 1 parameter but 2 parameters were supplied
LL | | }
   | |_^ expected 1 parameter

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0061`.
