error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:6:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:9:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:12:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: attribute must be applied to a `static` variable
  --> $DIR/used.rs:15:1
   |
LL | #[used] //~ ERROR attribute must be applied to a `static` variable
   | ^^^^^^^

error: aborting due to 4 previous errors

