error[E0426]: use of undeclared label `'a`
  --> $DIR/while-label-break-condition.rs:14:21
   |
LL |     'a: while break 'a {} //~ ERROR: use of undeclared label
   |                     ^^ undeclared label `'a`

error[E0426]: use of undeclared label `'b`
  --> $DIR/while-label-break-condition.rs:15:21
   |
LL |     'b: while break 'b {}; //~ ERROR: use of undeclared label
   |                     ^^ undeclared label `'b`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0426`.
