warning: irrefutable while-let pattern
  --> $DIR/while-let.rs:16:13
   |
LL |               while let $p = $e $b
   |               ^^^^^
...
LL | /     foo!(a, 1, { //~ warning irrefutable while-let
LL | |         println!("irrefutable pattern");
LL | |     });
   | |_______- in this macro invocation
   |
   = note: #[warn(irrefutable_let_patterns)] on by default

warning: irrefutable while-let pattern
  --> $DIR/while-let.rs:16:13
   |
LL |               while let $p = $e $b
   |               ^^^^^
...
LL | /     bar!(a, 1, { //~ warning irrefutable while-let
LL | |         println!("irrefutable pattern");
LL | |     });
   | |_______- in this macro invocation

warning: irrefutable while-let pattern
  --> $DIR/while-let.rs:34:5
   |
LL | /     while let a = 1 { //~ warning irrefutable while-let
LL | |         println!("irrefutable pattern");
LL | |     }
   | |_____^

