warning: not reporting region error due to -Znll
  --> $DIR/propagate-approximated-to-empty.rs:41:9
   |
41 |         demand_y(x, y, x.get())
   |         ^^^^^^^^^^^^^^^^^^^^^^^

error: free region `'_#6r` does not outlive free region `'_#4r`
  --> $DIR/propagate-approximated-to-empty.rs:41:21
   |
41 |         demand_y(x, y, x.get())
   |                     ^

note: External requirements
  --> $DIR/propagate-approximated-to-empty.rs:39:47
   |
39 |       establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
   |  _______________________________________________^
40 | |         // Only works if 'x: 'y:
41 | |         demand_y(x, y, x.get())
42 | |         //~^ WARN not reporting region error due to -Znll
43 | |         //~| ERROR free region `'_#6r` does not outlive free region `'_#4r`
44 | |     });
   | |_____^
   |
   = note: defining type: [closure@$DIR/propagate-approximated-to-empty.rs:39:47: 44:6<i16, for<'r, 's, 't0, 't1, 't2> extern "rust-call" fn((&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r(7714))) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's(7715))) &'_#1r u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't0(7716))) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't1(9540))) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't2(9541))) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's(7715))) u32>))>]
   = note: number of external vids: 2

note: No external requirements
  --> $DIR/propagate-approximated-to-empty.rs:38:1
   |
38 | / fn supply<'a, 'b>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>) {
39 | |     establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
40 | |         // Only works if 'x: 'y:
41 | |         demand_y(x, y, x.get())
...  |
44 | |     });
45 | | }
   | |_^
   |
   = note: defining type: for<'a, 'b> fn(std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(1:16), 'a(90))) u32>, std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(1:17), 'b(92))) u32>) {supply}

error: aborting due to previous error

