warning: not reporting region error due to nll
  --> $DIR/region-lbr-named-does-not-outlive-static.rs:20:5
   |
LL |     &*x
   |     ^^^

error: free region `ReFree(DefId(0/0:3 ~ region_lbr_named_does_not_outlive_static[317d]::foo[0]), BrNamed(crate0:DefIndex(1:9), 'a))` does not outlive free region `ReStatic`
  --> $DIR/region-lbr-named-does-not-outlive-static.rs:18:1
   |
LL | / fn foo<'a>(x: &'a u32) -> &'static u32 {
LL | |         //~^ ERROR does not outlive free region
LL | |     &*x
LL | |         //~^ WARN not reporting region error due to nll
LL | | }
   | |_^

error: aborting due to previous error

