warning: constant evaluation error: attempt to shift left with overflow
  --> $DIR/E0080.rs:12:9
   |
LL |     X = (1 << 500), //~ ERROR E0080
   |         ^^^^^^^^^^
   |
   = note: #[warn(const_err)] on by default

error[E0080]: constant evaluation error
  --> $DIR/E0080.rs:12:9
   |
LL |     X = (1 << 500), //~ ERROR E0080
   |         ^^^^^^^^^^ attempt to shift left with overflow

warning: constant evaluation error: attempt to divide by zero
  --> $DIR/E0080.rs:14:9
   |
LL |     Y = (1 / 0) //~ ERROR E0080
   |         ^^^^^^^

error[E0080]: constant evaluation error
  --> $DIR/E0080.rs:14:9
   |
LL |     Y = (1 / 0) //~ ERROR E0080
   |         ^^^^^^^ attempt to divide by zero

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0080`.
