error[E0261]: use of undeclared lifetime name `'a`
  --> $DIR/E0261.rs:11:12: in fn foo
   |
LL | fn foo(x: &'a str) { } //~ ERROR E0261
   |            ^^ undeclared lifetime

error[E0261]: use of undeclared lifetime name `'a`
  --> $DIR/E0261.rs:15:9: in struct Foo
   |
LL |     x: &'a str, //~ ERROR E0261
   |         ^^ undeclared lifetime

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0261`.
