error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> $DIR/sized-yield.rs:17:26: in fn main
   |
LL |      let mut gen = move || { //~ ERROR the trait bound `str: std::marker::Sized` is not satisfied
   |  __________________________^
LL | |        yield s[..];
LL | |    };
   | |____^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: the yield type of a generator must have a statically known size

error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> $DIR/sized-yield.rs:20:17: in fn main
   |
LL |    unsafe { gen.resume(); } //~ ERROR the trait bound `str: std::marker::Sized` is not satisfied
   |                 ^^^^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
