error[E0659]: `m` is ambiguous
  --> $DIR/macros.rs:38:5
   |
LL |     m!(); //~ ERROR ambiguous
   |     ^
   |
note: `m` could refer to the name defined here
  --> $DIR/macros.rs:36:5
   |
LL |     macro_rules! m { () => {} }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the name imported here
  --> $DIR/macros.rs:37:9
   |
LL |     use two_macros::m;
   |         ^^^^^^^^^^^^^

error[E0659]: `m` is ambiguous
  --> $DIR/macros.rs:16:5
   |
LL |     m! { //~ ERROR ambiguous
   |     ^
   |
note: `m` could refer to the name imported here
  --> $DIR/macros.rs:17:13
   |
LL |         use foo::m;
   |             ^^^^^^
note: `m` could also refer to the name imported here
  --> $DIR/macros.rs:15:9
   |
LL |     use two_macros::*;
   |         ^^^^^^^^^^^^^
   = note: macro-expanded macro imports do not shadow

error[E0659]: `m` is ambiguous
  --> $DIR/macros.rs:29:9
   |
LL |         m! { //~ ERROR ambiguous
   |         ^
   |
note: `m` could refer to the name imported here
  --> $DIR/macros.rs:30:17
   |
LL |             use two_macros::n as m;
   |                 ^^^^^^^^^^^^^^^^^^
note: `m` could also refer to the name imported here
  --> $DIR/macros.rs:22:9
   |
LL |     use two_macros::m;
   |         ^^^^^^^^^^^^^
   = note: macro-expanded macro imports do not shadow

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0659`.
