error[E0659]: `panic` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:33:5
   |
LL |     panic!(); //~ ERROR `panic` is ambiguous
   |     ^^^^^
   |
note: `panic` could refer to the name defined here
  --> $DIR/shadow_builtin_macros.rs:30:9
   |
LL |         macro_rules! panic { () => {} }
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |     } }
LL |     m!();
   |     ----- in this macro invocation
   = note: `panic` is also a builtin macro
   = note: macro-expanded macros do not shadow

error[E0659]: `panic` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:15:14
   |
LL |     fn f() { panic!(); } //~ ERROR ambiguous
   |              ^^^^^
   |
note: `panic` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:14:9
   |
LL |     use foo::*;
   |         ^^^^^^
   = note: `panic` is also a builtin macro
   = note: consider adding an explicit import of `panic` to disambiguate

error[E0659]: `panic` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:20:14
   |
LL |     fn f() { panic!(); } //~ ERROR ambiguous
   |              ^^^^^
   |
note: `panic` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:19:26
   |
LL |     ::two_macros::m!(use foo::panic;);
   |                          ^^^^^^^^^^
   = note: `panic` is also a builtin macro
   = note: macro-expanded macro imports do not shadow

error[E0659]: `n` is ambiguous
  --> $DIR/shadow_builtin_macros.rs:49:5
   |
LL |     n!(); //~ ERROR ambiguous
   |     ^
   |
note: `n` could refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:48:9
   |
LL |     use bar::*;
   |         ^^^^^^
note: `n` could also refer to the name imported here
  --> $DIR/shadow_builtin_macros.rs:36:13
   |
LL | #[macro_use(n)]
   |             ^
   = note: consider adding an explicit import of `n` to disambiguate

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0659`.
