fn main() -> Result<(), ()> {
    a(b(c(d(e(
        //~^ ERROR cannot find function `a`
        //~| ERROR cannot find function `b`
        //~| ERROR cannot find function `c`
        //~| ERROR cannot find function `d`
        //~| ERROR cannot find function `e`
        z????????????????????????????????????????????????????????????????????????????????????????
        ?????????????????????????????????????????????????????????????????????????????????????????
        ??????????????????????????????????????????????????????????????????
        //~^^^ ERROR cannot find value `z`
    )))))
}
