warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:16:13
   |
LL |               if let $p = $e $b
   |               ^^
...
LL | /     foo!(a, 1, { //~ warning irrefutable if-let
LL | |         println!("irrefutable pattern");
LL | |     });
   | |_______- in this macro invocation
   |
   = note: #[warn(irrefutable_let_patterns)] on by default

warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:16:13
   |
LL |               if let $p = $e $b
   |               ^^
...
LL | /     bar!(a, 1, { //~ warning irrefutable if-let
LL | |         println!("irrefutable pattern");
LL | |     });
   | |_______- in this macro invocation

warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:34:5
   |
LL | /     if let a = 1 { //~ warning irrefutable if-let
LL | |         println!("irrefutable pattern");
LL | |     }
   | |_____^

warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:38:5
   |
LL | /     if let a = 1 { //~ warning irrefutable if-let
LL | |         println!("irrefutable pattern");
LL | |     } else if true {
LL | |         println!("else-if in irrefutable if-let");
LL | |     } else {
LL | |         println!("else in irrefutable if-let");
LL | |     }
   | |_____^

warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:48:12
   |
LL |       } else if let a = 1 { //~ warning irrefutable if-let
   |  ____________^
LL | |         println!("irrefutable pattern");
LL | |     }
   | |_____^

warning: irrefutable if-let pattern
  --> $DIR/if-let.rs:54:12
   |
LL |       } else if let a = 1 { //~ warning irrefutable if-let
   |  ____________^
LL | |         println!("irrefutable pattern");
LL | |     }
   | |_____^

