error: incorrect close delimiter: `}`
  --> $DIR/token-error-correct-3.rs:28:9
   |
LL |         if !is_directory(path.as_ref()) { //~ ERROR: cannot find function `is_directory`
   |                                         - close delimiter possibly meant for this
LL |             callback(path.as_ref();
   |                     - un-closed delimiter
...
LL |         } else {
   |         ^ incorrect close delimiter

error: expected one of `)`, `,`, `.`, `?`, or an operator, found `;`
  --> $DIR/token-error-correct-3.rs:24:35
   |
LL |             callback(path.as_ref();
   |                     -             -
   |                     |             |
   |                     |             expected one of `)`, `,`, `.`, `?`, or an operator here
   |                     |             help: ...the missing `)` may belong here
   |                     if you meant to close this...

error: expected expression, found `;`
  --> $DIR/token-error-correct-3.rs:24:35
   |
LL |             callback(path.as_ref();
   |                                   ^ expected expression

error[E0425]: cannot find function `is_directory` in this scope
  --> $DIR/token-error-correct-3.rs:23:13
   |
LL |         if !is_directory(path.as_ref()) { //~ ERROR: cannot find function `is_directory`
   |             ^^^^^^^^^^^^ not found in this scope

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0425`.
