error[E0261]: use of undeclared lifetime name `'test`
  --> $DIR/no_in_band_in_struct.rs:15:9: in struct Foo
   |
LL |     x: &'test u32, //~ ERROR undeclared lifetime
   |         ^^^^^ undeclared lifetime

error[E0261]: use of undeclared lifetime name `'test`
  --> $DIR/no_in_band_in_struct.rs:19:10: in enum Bar
   |
LL |     Baz(&'test u32), //~ ERROR undeclared lifetime
   |          ^^^^^ undeclared lifetime

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0261`.
