error[E0308]: mismatched types
  --> $DIR/variance-invariant-self-trait-match.rs:12:5
   |
LL |     impls_get::<&'min G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 9:21...
  --> $DIR/variance-invariant-self-trait-match.rs:9:21
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 9:27
  --> $DIR/variance-invariant-self-trait-match.rs:9:27
   |
LL | fn get_min_from_max<'min, 'max, G>()
   |                           ^^^^

error[E0308]: mismatched types
  --> $DIR/variance-invariant-self-trait-match.rs:18:5
   |
LL |     impls_get::<&'max G>(); //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Get`
              found type `Get`
note: the lifetime 'min as defined on the function body at 15:21...
  --> $DIR/variance-invariant-self-trait-match.rs:15:21
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                     ^^^^
note: ...does not necessarily outlive the lifetime 'max as defined on the function body at 15:27
  --> $DIR/variance-invariant-self-trait-match.rs:15:27
   |
LL | fn get_max_from_min<'min, 'max, G>()
   |                           ^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
