error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:111:13
   |
LL |     o1.set0(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o2` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `o3` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:112:13
   |
LL |     o1.set1(&o3); //~ ERROR `o3` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o3` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:113:13
   |
LL |     o2.set0(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o2` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `o3` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:114:13
   |
LL |     o2.set1(&o3); //~ ERROR `o3` does not live long enough
   |             ^^^ borrowed value does not live long enough
...
LL | }
   | - `o3` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `o1` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:115:13
   |
LL |     o3.set0(&o1); //~ ERROR `o1` does not live long enough
   |             ^^^ borrowed value does not live long enough
LL |     o3.set1(&o2); //~ ERROR `o2` does not live long enough
LL | }
   | - `o1` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: `o2` does not live long enough
  --> $DIR/dropck_trait_cycle_checked.rs:116:13
   |
LL |     o3.set1(&o2); //~ ERROR `o2` does not live long enough
   |             ^^^ borrowed value does not live long enough
LL | }
   | - `o2` dropped here while still borrowed
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0597`.
