//! [pointer::add]
//~^ ERROR: experimental
//~| WARNING will not have documentation generated
//! [pointer::wrapping_add]
//~^ ERROR: experimental
//~| WARNING will not have documentation generated
//! [pointer] // This is explicitly allowed
//! [reference] // This is explicitly allowed
