error: unexpected generic arguments in path
  --> $DIR/visibility-ty-params.rs:16:6
   |
LL | m!{ S<u8> } //~ ERROR unexpected generic arguments in path
   |      ^^^^

error: unexpected generic arguments in path
  --> $DIR/visibility-ty-params.rs:19:10
   |
LL |     m!{ m<> } //~ ERROR unexpected generic arguments in path
   |          ^^

error: aborting due to 2 previous errors

