error[E0119]: conflicting implementations of trait `std::ops::Deref` for type `std::cell::Ref<'_, _>`:
  --> $DIR/issue-28981.rs:15:1
   |
LL | impl<Foo> Deref for Foo { } //~ ERROR must be used
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<'b, T> std::ops::Deref for std::cell::Ref<'b, T>
             where T: ?Sized;

error[E0210]: type parameter `Foo` must be used as the type parameter for some local type (e.g. `MyStruct<Foo>`)
  --> $DIR/issue-28981.rs:15:1
   |
LL | impl<Foo> Deref for Foo { } //~ ERROR must be used
   | ^^^^^^^^^^^^^^^^^^^^^^^ type parameter `Foo` must be used as the type parameter for some local type
   |
   = note: only traits defined in the current crate can be implemented for a type parameter

error: aborting due to 2 previous errors

Some errors occurred: E0119, E0210.
For more information about an error, try `rustc --explain E0119`.
