error: macros that expand to items must be delimited with braces or followed by a semicolon
  --> $DIR/macros-no-semicolon-items.rs:1:17
   |
LL | macro_rules! foo()
   |                 ^^
help: change the delimiters to curly braces
   |
LL | macro_rules! foo {}  //~ ERROR semicolon
   |                  ^^
help: add a semicolon
   |
LL | macro_rules! foo();  //~ ERROR semicolon
   |                   ^

error: macros that expand to items must be delimited with braces or followed by a semicolon
  --> $DIR/macros-no-semicolon-items.rs:8:5
   |
LL |   bar!( //~ ERROR semicolon
   |  _____^
LL | |     blah
LL | |     blah
LL | |     blah
LL | | )
   | |_^
help: change the delimiters to curly braces
   |
LL | bar! { //~ ERROR semicolon
LL |     blah
LL |     blah
LL |     blah
LL | }
   |
help: add a semicolon
   |
LL | );
   |  ^

error: unexpected end of macro invocation
  --> $DIR/macros-no-semicolon-items.rs:1:1
   |
LL | macro_rules! foo()  //~ ERROR semicolon
   | ^^^^^^^^^^^^^^^^^^ missing tokens in macro arguments

error: aborting due to 3 previous errors

