error: struct `m::Priv` is private
  --> $DIR/private-inferred-type-2.rs:16:5
   |
LL |     m::Pub::get_priv; //~ ERROR struct `m::Priv` is private
   |     ^^^^^^^^^^^^^^^^ private

error: struct `m::Priv` is private
  --> $DIR/private-inferred-type-2.rs:17:5
   |
LL |     m::Pub::static_method; //~ ERROR struct `m::Priv` is private
   |     ^^^^^^^^^^^^^^^^^^^^^ private

error: struct `ext::Priv` is private
  --> $DIR/private-inferred-type-2.rs:18:5
   |
LL |     ext::Pub::static_method; //~ ERROR struct `ext::Priv` is private
   |     ^^^^^^^^^^^^^^^^^^^^^^^ private

error: aborting due to 3 previous errors

