error: struct `m::Priv` is private
  --> $DIR/private-inferred-type-1.rs:16:5
   |
LL |     [].arr0_secret(); //~ ERROR struct `m::Priv` is private
   |     ^^^^^^^^^^^^^^^^ private

error: struct `m::Priv` is private
  --> $DIR/private-inferred-type-1.rs:17:5
   |
LL |     None.ty_param_secret(); //~ ERROR struct `m::Priv` is private
   |     ^^^^^^^^^^^^^^^^^^^^^^ private

error: aborting due to 2 previous errors

