error: `$ty:ty` is followed by `<`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:5:13
   |
LL |     ($ty:ty <) => (); //~ ERROR `$ty:ty` is followed by `<`, which is not allowed for `ty`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$ty:ty` is followed by `<`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:6:13
   |
LL |     ($ty:ty < foo ,) => (); //~ ERROR `$ty:ty` is followed by `<`, which is not allowed for `ty`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$pa:pat` is followed by `>`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:12:14
   |
LL |     ($pa:pat >) => (); //~ ERROR `$pa:pat` is followed by `>`, which is not allowed for `pat`
   |              ^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$pa:pat` is followed by `$pb:pat`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:14:14
   |
LL |     ($pa:pat $pb:pat $ty:ty ,) => ();
   |              ^^^^^^^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$pb:pat` is followed by `$ty:ty`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:14:22
   |
LL |     ($pa:pat $pb:pat $ty:ty ,) => ();
   |                      ^^^^^^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$ty:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:17:13
   |
LL |     ($ty:ty -) => (); //~ ERROR `$ty:ty` is followed by `-`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$b:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:18:19
   |
LL |     ($a:ty, $b:ty -) => (); //~ ERROR `$b:ty` is followed by `-`
   |                   ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$ty:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:19:7
   |
LL |     ($($ty:ty)-+) => (); //~ ERROR `$ty:ty` is followed by `-`, which is not allowed for `ty`
   |       ^^^^^^^^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$a:expr` is followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:20:17
   |
LL |     ( $a:expr $($b:tt)* ) => { };
   |                 ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

error: `$a:expr` is followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:22:21
   |
LL |     ( $($a:expr)* $($b:tt)* ) => { };
   |                     ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

error: `$a:expr` may be followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:26:22
   |
LL |     ( $($a:expr),* $($b:tt)* ) => { };
   |                      ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

warning: `$a:expr` is followed (through repetition) by itself, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:22:9
   |
LL |     ( $($a:expr)* $($b:tt)* ) => { };
   |         ^^^^^^^ this fragment is followed by itself without a valid separator
   |
note: lint level defined here
  --> $DIR/macro-input-future-proofing.rs:2:9
   |
LL | #![warn(incorrect_macro_fragment_repetition)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #56575 <https://github.com/rust-lang/rust/issues/56575>
   = note: allowed there are: `;`, `=>` or `,`
help: add a valid separator for the repetition to be unambiguous, for example
   |
LL |     ( $($a:expr);* $($b:tt)* ) => { };
   |                 ^

error: aborting due to 11 previous errors

